internal/proxy/util.go

```diff
@@ -609,6 +610,87 @@ func validateSchema(coll *schemapb.CollectionSchema) error {
 	return nil
 }
 
+func validateAndFillFunction(coll *schemapb.CollectionSchema) error {
+	nameMap := lo.SliceToMap(coll.GetFields(), func(field *schemapb.FieldSchema) (string, *schemapb.FieldSchema) {
+		return field.GetName(), field
+	})
+	// validate function
+	for _, function := range coll.GetFunctions() {
+		inputFields := []*schemapb.FieldSchema{}
+		for _, name := range function.GetInputFieldNames() {
+			inputField, ok := nameMap[name]
+			if !ok {
+				return fmt.Errorf("function input field not found %s", function.InputFieldNames)
+			}
+			inputFields = append(inputFields, inputField)
+		}
+
+		err := checkFunctionInputField(function, inputFields)
+		if err != nil {
+			return err
+		}
+
+		outputFields := make([]*schemapb.FieldSchema, len(function.GetOutputFieldNames()))
+		missingFields := []int{}
+		for i, name := range function.GetOutputFieldNames() {
+			outputField, ok := nameMap[name]
+			if !ok {
+				outputFields[i] = &schemapb.FieldSchema{
+					Name: name,
+				}
+				missingFields = append(missingFields, i)
+				continue
+			}
+			outputFields[i] = outputField
+		}
+
+		err = checkAndFillFunctionOutputField(function, coll, outputFields, missingFields)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
```

Comment:
I think we need to sort fields in coll:

* all user defined, non function output fields should come first, sorted by how the user defines them in SDK
* all function output fields should be after other fields. they should first be sorted by function order, then sorted by how they are defined in SDK

Reply:
```python
schema = milvus_client.create_schema() 
schema.add_field("id", DataType.INT64, is_primary=True, auto_id=True) 
schema.add_field("text", DataType.VARCHAR, max_length=1000)

bm25_function_1 = Function(
   name="text_bm25_emb_1", 
   inputs="text", 
   outputs="sparse_1",
   function_type=FunctionType.BM25, 
   params={"k1": 1.2, "b": 0.75}, 
)

bm25_function_2 = Function(
   name="text_bm25_emb_2", 
   inputs="text", 
   outputs=FieldSchema("sparse_2", SPARSE_FLOAT_VECTOR),
   function_type=FunctionType.BM25, 
   params={"k1": 1.2, "b": 0.75}, 
)

schema.add_function(bm25_function_1)
schema.add_function(bm25_function_2)
```

in create collection proto, there will be 3 fields: `id, text, sparse_2`. and when we `checkAndFillFunctionOutputField` on `sparse_1`, we append it: the order becomes `id, text, sparse_2, sparse_1`, but we wanted `id, text, sparse_1, sparse_2`

Reply:
also, when assigning field ids, let's offset function output fields by maybe 1000(current max number of fields is 64)? so that in the future when the user adds fields by altering the collection, function output fields still have larger field ids.